auth/auth.go

```diff
@@ -36,14 +36,19 @@ func NewClient(clientID string, clientSecret string, issuerURL string, httpClien
 	// Need to wait for identity service to become available.
 	var provider *oidc.Provider
 	var err error
+	attempt := 1
 	for {
 		provider, err = oidc.NewProvider(ctx, issuerURL)
 		if err != nil {
 			log.Errorf("failed to query provider %q: %v", issuerURL, err)
-			time.Sleep(5 * time.Second)
+			time.Sleep(time.Duration(5) * time.Duration(attempt) * time.Second)
 		} else {
 			break
 		}
+		attempt++
+		if attempt > 3 {
+			log.Fatal("Too many attempts... server not responding")
```

Comment:
It's best to not terminate from a client library. Leave it up to the caller to decide how it's going to handle failure. Here it might be best to extend the function to return the error.